fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/impl/registry/sync/FabricRegistryInit.java

```diff
@@ -0,0 +1,158 @@
+		RegistryAttributeHolder.get(Registry.BIOME)
+				.addAttribute(RegistryAttribute.SYNCED)
+				.addAttribute(RegistryAttribute.PERSISTED);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.BLOCK_STATE_PROVIDER_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.BLOCK_PLACER_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.FOLIAGE_PLACER_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.TRUNK_PLACER_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.TREE_DECORATOR_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.FEATURE_SIZE_TYPE);
+
+		// RegistryAttribute.SYNCED in ParticleS2CPacket
+		RegistryAttributeHolder.get(Registry.PARTICLE_TYPE)
+				.addAttribute(RegistryAttribute.SYNCED);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.BIOME_SOURCE_TYPE);
+
+		// Serialised by string, doesnt seem to be RegistryAttribute.SYNCED
+		RegistryAttributeHolder.get(Registry.CHUNK_GENERATOR_TYPE);
+
+		// RegistryAttribute.SYNCED in GameJoinS2CPacket and PlayerRespawnS2CPacket. Entities and maps also use the id when saving.
+		RegistryAttributeHolder.get(Registry.DIMENSION_TYPE)
+				.addAttribute(RegistryAttribute.SYNCED)
+				.addAttribute(RegistryAttribute.PERSISTED);
+
+		// RegistryAttribute.SYNCED in PaintingSpawnS2CPacket
+		RegistryAttributeHolder.get(Registry.PAINTING_MOTIVE)
+				.addAttribute(RegistryAttribute.SYNCED);
+
+		// Doesnt seem to be RegistryAttribute.SYNCED or saved, STAT_TYPE seems to handle the syncing.
+		RegistryAttributeHolder.get(Registry.CUSTOM_STAT);
+
+		RegistryAttributeHolder.get(Registry.CHUNK_STATUS);
+
+		RegistryAttributeHolder.get(Registry.STRUCTURE_FEATURE);
+
+		RegistryAttributeHolder.get(Registry.STRUCTURE_PIECE);
+
+		RegistryAttributeHolder.get(Registry.RULE_TEST);
+
+		RegistryAttributeHolder.get(Registry.POS_RULE_TEST);
+
+		RegistryAttributeHolder.get(Registry.STRUCTURE_PROCESSOR);
+
+		RegistryAttributeHolder.get(Registry.STRUCTURE_POOL_ELEMENT);
+
+		// RegistryAttribute.SYNCED in OpenScreenS2CPacket
+		RegistryAttributeHolder.get(Registry.SCREEN_HANDLER)
+				.addAttribute(RegistryAttribute.SYNCED);
+
+		RegistryAttributeHolder.get(Registry.RECIPE_TYPE);
+
+		RegistryAttributeHolder.get(Registry.RECIPE_SERIALIZER);
+
+		RegistryAttributeHolder.get(Registry.ATTRIBUTES);
+
+		// RegistryAttribute.SYNCED in StatisticsS2CPacket
+		RegistryAttributeHolder.get(Registry.STAT_TYPE)
+				.addAttribute(RegistryAttribute.SYNCED);
+
+		RegistryAttributeHolder.get(Registry.VILLAGER_TYPE);
+
+		RegistryAttributeHolder.get(Registry.VILLAGER_PROFESSION);
```

Comment:
I assume the fact this is used in a data tracker isn't a real issue with syncing. (I think it calls getRawId when being queried to track) 

This is part of the VillagerDataContainer but that stores this as an identifier.

Something similar with the Villager professions.

Reply:
Looking at this it needs to be synced as its written to a packet using the raw id.